### PR TITLE
Chore/timeline

### PR DESCRIPTION
**Pull request by tejohnso**
Timeline processing logic.
@fjvallarino / @alex-deaconu  please review

### CODE REVIEW

**Comment by fjvallarino** [comment]
:+1:

File: app/player/schedule/timeline-parser.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+      return true;
+    }
+  };
+
+  function err(msg) {throw new Error(msg); }
+
+  function checkStartEndDateRange() {
+    if (!timeline.hasOwnProperty("timeDefined")) {err("time defined"); }
+    if (timeline.timeDefined === false) {err("time defined false"); }
```
Comment (alex-deaconu) [inline app/player/schedule/timeline-parser.js]: @tejohnso `timeDefined` indicates whether or not a timeline is defined. If it is `false`, `canPlay()` should always return `true`. `true` indicates a user has selected a timeline. Thanks.

File: app/player/schedule/timeline-parser.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    if (!timeline.hasOwnProperty("startDate")) {err("start date"); }
```
Comment (alex-deaconu) [inline app/player/schedule/timeline-parser.js]: @tejohnso I don't believe `startDate` should be required. If it is not there it should be ignored, and the check should succeed. Thoughts? Thanks.

File: app/player/schedule/timeline-parser.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+      startDate = getDateComponent(timelineObject.startDate);
+      endDate = getDateComponent(timelineObject.endDate);
+
+      recurrenceFrequency = timeline.recurrenceFrequency;
+      if (recurrenceFrequency < 1) {recurrenceFrequency = 1;}
+
+      checkStartEndDateRange();
+      checkStartEndTimeRange();
+      checkRecurrence[timeline.recurrenceType]();
```
Comment (alex-deaconu) [inline app/player/schedule/timeline-parser.js]: @tejohnso should probably check if a correct `recurrenceType` is set (or ignore this check). Thoughts? Thanks.

**Comment by alex-deaconu** [comment]
@tejohnso reviewed. Compared to current functionality in the RVA/Viewer, and found a couple of differences. Otherwise looks good. Thanks.

**Comment by tejohnso** [comment]
@alex-deaconu Updated at a383697

**Comment by alex-deaconu** [comment]
@tejohnso looks good :+1: . Thanks!

**Pull request merged**